src/graph/graphencodecontext.h

```diff
@@ -0,0 +1,117 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#pragma once
+
+#include "stdlib.h"
+#include "stdint.h"
+#include "stdbool.h"
+#include "../util/datablock/datablock.h"
+#include "../../deps/GraphBLAS/Include/GraphBLAS.h"
+#include "rax.h"
+
+// Represent a graph encoding phase.
+typedef enum {
+	RESET,
+	NODES,
+	DELETED_NODES,
+	EDGES,
+	DELETED_EDGES,
+	GRAPH_SCHEMA
+} EncodePhase;
+
+// A struct that maintains the state of a graph encoding to RDB or encode from RDB.
+typedef struct {
+	uint64_t keys_processed;                    // Count the number of procssed graph keys.
+	rax *keys;                                  // The name of the meta keys representing the graph.
```

Comment:
Can we replace this rax with a simple total `keys_to_process`? `_CreateGraphMetaKeys` will create the appropriate N keys; we don't actually care what their names are.

If I understand right we can remove this rax and a large amount of naming logic for meta keys; we're done when `keys_processed == keys_to_process`.

Reply:
done